crates/modelardb_server/src/storage/uncompressed_data_manager.rs

```diff
@@ -277,12 +282,23 @@ impl UncompressedDataManager {
                 .map(|array| array.value(index).to_string())
                 .collect();
 
-            let tag_hash = self
+            let (tag_hash, tag_hash_is_saved) = self
                 .table_metadata_manager
                 .lookup_or_compute_tag_hash(&model_table_metadata, &tag_values)
                 .await
                 .map_err(|error| format!("Tag hash could not be saved: {error}"))?;
 
+            // If the server was started with a manager, transfer the tag hash metadata if it was
```

Comment:
Will this not create false positives as the files with the data have yet to be transferred? If this is the intended behavior maybe extend the comment to document it.

Reply:
It was my understanding that when we use the tag metadata during querying, we retrieve the tag values based on the tag hash part of the univariate id in the compressed data. This would mean that it does not matter if there are extra tag hash keys in the hash map we retrieve from. You have a better understanding of querying so please correct me if i am wrong.

Furthermore, if this was a problem when transferring to the manager, would it also not be a problem locally? Since even if running without a manager, we save the tag hash metadata immediately and save the compressed data later.

Reply:
No I agree that it is not a problem to transfer tag information before the files containing the associated data, it is only a problem if we transfer file metadata before the files as the query engine then will try to read files from the object store that does not exist and crash. My comment was mainly men't as a question to confirm that the tags are purposely transferred before the files with the associated data and to suggest that a comment be added to document this.

Reply:
Fixed.